src/test/java/unit/BqDatasetLifetime.java

```diff
@@ -0,0 +1,227 @@
+    assertEquals(tableLifetime, defaultLifetimeMap.get(DefaultLifetimeEntity.TABLE));
+  }
+
+  /**
+   * Helper method to get the name of the dataset that should be created by the test method (derived
+   * from the passed TestInfo instance).
+   */
+  private String getDatasetName(TestInfo testInfo) {
+    return testInfo.getTestMethod().orElseThrow().getName();
+  }
+
+  /** Creates a BQ Dataset with the given name and any passed lifetime arguments. */
+  private UFBqDataset createDatasetWithLDefaultLifetimes(
+      String name, String... defaultLifetimeArguments) throws JsonProcessingException {
+
+    String datasetId = randomDatasetId();
+
+    List<String> arguments =
+        new ArrayList<>(
+            Arrays.asList(
+                "resource", "create", "bq-dataset", "--name=" + name, "--dataset-id=" + datasetId));
+
+    arguments.addAll(Arrays.asList(defaultLifetimeArguments));
+
+    return TestCommand.runAndParseCommandExpectSuccess(
+        UFBqDataset.class, arguments.toArray(new String[0]));
+  }
+
+  /** Updates a dataset with a given name using the passed arguments. */
+  private void updateDatasetWithDefaultLifetimes(String name, String... defaultLifetimeArguments) {
+    List<String> arguments =
+        new ArrayList<>(Arrays.asList("resource", "update", "bq-dataset", "--name=" + name));
+
+    arguments.addAll(Arrays.asList(defaultLifetimeArguments));
+    TestCommand.runCommandExpectSuccess(arguments.toArray(new String[0]));
+  }
+
+  /** Enum to use as a key for a map of entity to lifetime values. */
+  private enum DefaultLifetimeEntity {
+    PARTITION,
+    TABLE
+  }
+
+  /**
+   * Use the Google BQ Client to get the configured default lifetime for the passed dataset
+   * descriptor and return it as a map of entity type (partition or table) to its configured default
+   * lifetime in seconds.
+   */
+  private Map<DefaultLifetimeEntity, Integer> getDefaultLifetimes(UFBqDataset bqDataset)
+      throws IOException {
+
+    DatasetId datasetId = DatasetId.of(bqDataset.projectId, bqDataset.datasetId);
+
+    Dataset dataset =
+        ExternalBQDatasets.getBQClient(workspaceCreator.getCredentialsWithCloudPlatformScope())
+            .getDataset(datasetId);
+
+    assertNotNull(dataset, "looking up dataset via BQ API succeeded");
+
+    // These methods return null if lifetime is not set, if they are set they need to be converted
+    // to integers representing seconds to be returned in the map, with 0 representing no lifetime
+    // set.  Both of these methods return milliseconds.
+    Long defaultPartitionExpirationMs = dataset.getDefaultPartitionExpirationMs();
+    Long defaultTableLifetime = dataset.getDefaultTableLifetime();
+
+    return Map.of(
+        DefaultLifetimeEntity.PARTITION,
+        defaultPartitionExpirationMs == null ? 0 : defaultPartitionExpirationMs.intValue() / 1000,
+        DefaultLifetimeEntity.TABLE,
+        defaultTableLifetime == null ? 0 : defaultTableLifetime.intValue() / 1000);
```

Comment:
optional style: this is a little too much logic in the test for my preference. Fine to have a helper method that retrieves the `Dataset`, but consider just asserting directly against `getDefaultPartionExpirationMs` and `getDefaultTableLifetime`, nulls and all.

Reply:
streamlined this checking by using a `Duration` and minimize hand-rolled conversion logic